### PR TITLE
usb: hid: Call the report proto callback function when the reset occurs

### PR DESCRIPTION
**Pull request by Chenhongren**
This commit changes the protocol callback function is invoked if the HID protocol changed due to reset.

### CODE REVIEW

**Review by tmon-nordic — COMMENTED** [review]
> The protocol is set as report protocol when the reset interrupt occurs. Meanwhile, the protocol callback function is invoked.

I only understood what you mean after reading the code. You should probably reword it to indicate that this change does call protocol change callback if the HID protocol changed due to reset.

**Comment by Chenhongren** [comment]
The description is fixed, thanks.

File: subsys/usb/device/class/hid/core.c [hunk 1]
```diff
@@ -354,13 +354,21 @@ static void hid_do_status_cb(struct hid_device_info *dev_data,
...
 		dev_data->suspended = false;
 #ifdef CONFIG_USB_HID_BOOT_PROTOCOL
+		const struct device *dev = dev_data->common.dev;
+		uint8_t prev = dev_data->protocol;
 		dev_data->protocol = HID_PROTOCOL_REPORT;
+		if (prev != HID_PROTOCOL_REPORT) {
+			if (dev_data->ops && dev_data->ops->protocol_change) {
+				dev_data->ops->protocol_change(dev, dev_data->protocol);
+			}
+		}
```
Comment (jfischer-no) [inline subsys/usb/device/class/hid/core.c:369]:
```suggestion
		const struct device *dev = dev_data->common.dev;
		uint8_t prev = dev_data->protocol;
		
		dev_data->protocol = HID_PROTOCOL_REPORT;
		if (prev != HID_PROTOCOL_REPORT) {
			if (dev_data->ops && dev_data->ops->protocol_change) {
				dev_data->ops->protocol_change(dev, dev_data->protocol);
			}
		}
```

Comment (Chenhongren) [reply subsys/usb/device/class/hid/core.c]: Fixed. Thank you!

**Review by jfischer-no — APPROVED** [review]

**Review by tmon-nordic — APPROVED** [review]
